docs/DeveloperGuide.md

```diff
@@ -3,36 +3,144 @@
+- **Categorization**: Transactions are categorized as inflow or outflow based on their type. This categorization enables users to differentiate between different types of financial activities, such as income and expenses.
+- **Formatting**: The transactions are presented in a formatted string, providing clear and concise information to the user. Each transaction includes its description, date, and time.
+
+#### Implementation Details:
+- The `TransactionManager#showLastNTransactions()` method is responsible for retrieving the last N transactions from the Transaction List.
+- The method iterates over the transaction list in reverse order, starting from the latest transaction and moving towards older transactions.
+- Transactions are categorized as inflows or outflows based on their instance type (Inflow or Outflow), and the relevant transactions are added to the output string accordingly.
+- The formatted string containing the transaction details is returned to the user for display.
+
+#### Alternatives Considered:
+During the development of the View Transaction History feature, several alternatives were considered to achieve the desired functionality. These alternatives included:
+
+- Implementing a more complex data structure for storing transaction history, such as a linked list. However, a simple ArrayList-based approach was chosen for its simplicity and efficiency in managing transaction data.
+- Providing additional filtering options, such as filtering transactions by date range or transaction category. While these options could definitely enhance the feature, they were deemed unnecessary for the initial implementation and may be considered for future iterations.
+
+### Save File feature:
+The `Storage#saveFile()` method will be called when the user decides to quit the application, the method will then get a string of all the transactions. This string will then be written to a file whose filename will be the username of the last user.
+
+#### Implementation Details:
+- The method first creates a FileWriter object.
+- It then calls `TransactionManager#toSave()`. 
+  - This method is similar to `toString()`, except that it creates a string that is more storage friendly.
+  - In this method, each transaction is returned as `name|amount|date|category\n`. 
+  - Each transaction is added line by line, after adding all transactions, the string is returned in the toSave() method.
+  - It is important to note that outflows are saved with a `-` in front of the amount. 
+- With the string generated in the `toSave()` method, the FileWriter will write it to the file before closing it.
+
+### Load File feature:
+The `Storage#loadFile()` method will be called after the user as logged in, it will then retrieve the transactions that were written in the past sessions, so that the user can continue using the finance manger.
+
+#### Implementation Details:
+- The method first creates a File object and a Transaction Manager object.
```

Comment:
Perhaps you could include a class diagram to demonstrate how the saveFile class works.